packages/dashboard/e2e/tests/utils/dragAndDrop.ts

```diff
@@ -0,0 +1,69 @@
+import { Locator, Page } from '@playwright/test';
+import zip from 'lodash/zip';
+import { isDefined } from '../../../src/util/isDefined';
+
+type Point = [number, number];
+
+const center = (box: { x: number; y: number; height: number; width: number }): [x: number, y: number] => [
+  box.x + box.width / 2,
+  box.y + box.height / 2,
+];
+
+const subdivide = (arr: Point[]) => {
+  if (arr.length <= 1) return arr;
+  const grouped: Point[][] = [];
+  for (let i = 0; i < arr.length - 1; i++) {
+    grouped.push([arr[i], arr[i + 1]]);
+  }
+  const divided = grouped.map(([a, b]: Point[]): Point => {
+    const [ax, ay] = a;
+    const [bx, by] = b;
+
+    const mx = (ax + bx) / 2;
+    const my = (ay + by) / 2;
+    return [mx, my];
+  });
+  return zip(arr, divided).flat().filter(isDefined);
+};
```

Comment:
It would be very difficult for me to update this function. Please consider decomposing and adding comments.